src/Data/Morpheus/Types/Internal/Value.hs

```diff
@@ -18,12 +18,56 @@ import qualified Data.Vector         as V (toList)
 import           GHC.Generics        (Generic)
 
 convertToJSONName :: Text -> Text
-convertToJSONName "type'" = "type"
```

Comment:
could we define do 


```haskell
isReserved:: Text -> Bool

convertToJSONName hsName
     | isReserved name && (lastChar == ''') = name
        where (name, lastChar) =  ...
     | otherwise = hsName

convertToHaskellName name 
    | isReserved name = name <> "'"
    | otherwise = name
```

or do you think it will decrease performance?

Reply:
would be also great if you mention it in `README.md`
and update `changelog.md`. thanks :)

Reply:
Well, you don't really know without benchmarks, do you? ;-)
horus/morpheus-graphql@3514312

This produced some interesting results:

My laptop with GHC-8.6.5:

```
morpheus-graphql> benchmarks
Running 1 benchmarks...
Benchmark convert-benchmarks: RUNNING...
benchmarking to-hs/def
time                 108.7 ns   (106.9 ns .. 111.0 ns)
                     0.997 R²   (0.995 R² .. 0.998 R²)
mean                 109.7 ns   (107.3 ns .. 113.3 ns)
std dev              10.03 ns   (6.831 ns .. 15.34 ns)
variance introduced by outliers: 89% (severely inflated)

benchmarking to-hs/def-kw
time                 66.07 ns   (64.70 ns .. 67.50 ns)
                     0.997 R²   (0.996 R² .. 0.998 R²)
mean                 66.65 ns   (65.43 ns .. 68.21 ns)
std dev              4.707 ns   (3.856 ns .. 5.805 ns)
variance introduced by outliers: 83% (severely inflated)

benchmarking to-hs/checkfirst
time                 132.4 ns   (130.8 ns .. 133.8 ns)
                     0.998 R²   (0.997 R² .. 0.999 R²)
mean                 132.8 ns   (130.5 ns .. 135.4 ns)
std dev              8.401 ns   (6.709 ns .. 10.93 ns)
variance introduced by outliers: 79% (severely inflated)

benchmarking to-hs/checkfirst-kw
time                 176.7 ns   (174.0 ns .. 179.6 ns)
                     0.998 R²   (0.997 R² .. 0.999 R²)
mean                 179.7 ns   (176.9 ns .. 183.0 ns)
std dev              9.854 ns   (7.954 ns .. 12.80 ns)
variance introduced by outliers: 74% (severely inflated)

benchmarking to-json/def
time                 153.5 ns   (150.7 ns .. 156.3 ns)
                     0.997 R²   (0.996 R² .. 0.998 R²)
mean                 156.9 ns   (153.5 ns .. 161.7 ns)
std dev              13.06 ns   (9.489 ns .. 19.30 ns)
variance introduced by outliers: 87% (severely inflated)

benchmarking to-json/def-kw
time                 167.6 ns   (164.1 ns .. 171.7 ns)
                     0.996 R²   (0.994 R² .. 0.998 R²)
mean                 170.7 ns   (166.9 ns .. 174.8 ns)
std dev              12.87 ns   (10.68 ns .. 15.72 ns)
variance introduced by outliers: 84% (severely inflated)

benchmarking to-json/checkfirst
time                 100.6 ns   (99.02 ns .. 102.2 ns)
                     0.997 R²   (0.996 R² .. 0.998 R²)
mean                 101.6 ns   (99.76 ns .. 103.8 ns)
std dev              6.766 ns   (5.473 ns .. 8.585 ns)
variance introduced by outliers: 81% (severely inflated)

benchmarking to-json/checkfirst-kw
time                 144.3 ns   (142.2 ns .. 146.1 ns)
                     0.998 R²   (0.997 R² .. 0.999 R²)
mean                 144.1 ns   (141.8 ns .. 148.0 ns)
std dev              10.29 ns   (7.499 ns .. 14.57 ns)
variance introduced by outliers: 83% (severely inflated)

Benchmark convert-benchmarks: FINISH
```

Another FreeBSD machine with GHC-8.4.4:

```
morpheus-graphql> benchmarks
Running 1 benchmarks...
Benchmark convert-benchmarks: RUNNING...
benchmarking to-hs/def
time                 84.76 ns   (84.63 ns .. 84.89 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 84.76 ns   (84.64 ns .. 84.94 ns)
std dev              515.6 ps   (366.1 ps .. 726.7 ps)

benchmarking to-hs/def-kw
time                 59.38 ns   (59.37 ns .. 59.38 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 59.38 ns   (59.37 ns .. 59.38 ns)
std dev              7.074 ps   (5.595 ps .. 9.434 ps)

benchmarking to-hs/checkfirst
time                 88.16 ns   (86.95 ns .. 89.51 ns)
                     0.999 R_   (0.998 R_ .. 0.999 R_)
mean                 88.12 ns   (87.12 ns .. 89.18 ns)
std dev              3.354 ns   (2.857 ns .. 3.998 ns)
variance introduced by outliers: 58% (severely inflated)

benchmarking to-hs/checkfirst-kw
time                 154.9 ns   (154.9 ns .. 155.0 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 155.0 ns   (155.0 ns .. 155.0 ns)
std dev              109.0 ps   (85.72 ps .. 130.2 ps)

benchmarking to-json/def
time                 83.91 ns   (83.90 ns .. 83.93 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 83.92 ns   (83.90 ns .. 83.95 ns)
std dev              60.25 ps   (27.99 ps .. 113.0 ps)

benchmarking to-json/def-kw
time                 100.7 ns   (100.7 ns .. 100.7 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 100.7 ns   (100.7 ns .. 100.8 ns)
std dev              72.27 ps   (59.28 ps .. 89.19 ps)

benchmarking to-json/checkfirst
time                 86.61 ns   (86.57 ns .. 86.65 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 86.66 ns   (86.61 ns .. 86.71 ns)
std dev              173.1 ps   (138.3 ps .. 225.9 ps)

benchmarking to-json/checkfirst-kw
time                 105.7 ns   (105.6 ns .. 105.8 ns)
                     1.000 R_   (1.000 R_ .. 1.000 R_)
mean                 105.7 ns   (105.6 ns .. 105.7 ns)
std dev              248.2 ps   (212.0 ps .. 291.4 ps)

Benchmark convert-benchmarks: FINISH
```

The same machine with GHC-8.6.5:

```
morpheus-graphql> benchmarks
Running 1 benchmarks...
Benchmark convert-benchmarks: RUNNING...
benchmarking to-hs/def
time                 86.11 ns   (86.10 ns .. 86.13 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 86.10 ns   (86.08 ns .. 86.11 ns) 
std dev              56.73 ps   (40.99 ps .. 76.78 ps) 

benchmarking to-hs/def-kw
time                 58.77 ns   (58.77 ns .. 58.78 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 58.77 ns   (58.77 ns .. 58.78 ns) 
std dev              6.227 ps   (4.963 ps .. 8.170 ps) 

benchmarking to-hs/checkfirst
time                 90.22 ns   (90.22 ns .. 90.22 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 90.23 ns   (90.22 ns .. 90.23 ns) 
std dev              15.86 ps   (13.27 ps .. 19.09 ps) 

benchmarking to-hs/checkfirst-kw
time                 154.4 ns   (154.3 ns .. 154.5 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 154.3 ns   (154.3 ns .. 154.4 ns) 
std dev              194.2 ps   (144.2 ps .. 300.5 ps) 

benchmarking to-json/def
time                 82.49 ns   (82.46 ns .. 82.51 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 82.50 ns   (82.48 ns .. 82.51 ns) 
std dev              53.35 ps   (41.55 ps .. 63.01 ps) 

benchmarking to-json/def-kw
time                 98.23 ns   (98.18 ns .. 98.27 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 98.17 ns   (98.13 ns .. 98.21 ns) 
std dev              134.6 ps   (112.6 ps .. 160.2 ps) 

benchmarking to-json/checkfirst
time                 86.39 ns   (86.35 ns .. 86.45 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 86.39 ns   (86.35 ns .. 86.45 ns) 
std dev              142.9 ps   (94.58 ps .. 199.7 ps) 

benchmarking to-json/checkfirst-kw
time                 106.8 ns   (106.8 ns .. 106.8 ns) 
                     1.000 R_   (1.000 R_ .. 1.000 R_) 
mean                 106.8 ns   (106.8 ns .. 106.8 ns) 
std dev              19.36 ps   (10.36 ps .. 35.55 ps) 

Benchmark convert-benchmarks: FINISH
```

Roughly, my conclusion would be: the naive implementation is doing fine most of the time, but you can tell from the outputs, the implementation with `isReserved` is OK, too, sometimes even does a bit better. (I guess upgrade to GHC-8.6.x would also be helpful)

Reply:
okey. then i think we should go for `isReserved`.

Reply:
Done.